polaris/utils/errors.py

```diff
@@ -37,3 +40,21 @@ class TestAccessError(Exception):
 
 class InvalidZarrChecksum(Exception):
     pass
+
+
+class PolarisCreateArtifactError(PolarisHubError):
+    SUGGEST_LOGIN_ON_CREATE = "Note: If you can confirm that you are authorized to perform this action, please call 'polaris login --overwrite' and try again. If this issue persists, please reach out to the Polaris team for support"
+
+    def __str__(self):
+        base_message = super().__str__()
```

Comment:
**Suggestion:** Instead of the `super.__str__()`, I think we should use the `__init__()` constructors. That way we benefit from any other functionality that may be based on `self.message`. I also don't think we need to define the messages as class variables, since we only use them in one spot. 

Something like: 

```py
class PolarisCreateArtifactError(PolarisHubError):
    def __init__(self): 
        message = self.format(
            "Note: If you can confirm that you are authorized to perform this action, " 
            "please call 'polaris login --overwrite' and try again.",
            [self.BOLD, self.YELLOW]
        )
        super().__init__(message)
```

In `PolarisHubError`, you could similarly add a prefix (and suffix) with the same approach.

Reply:
We use the `super.__str__()` because I thought we didn't want to explicitely pass in a `response` object. We need to use the `__str__` method in that case because the `self.__cause__` variable is not defined until `__init__` has run.

Reply:
Also, at least from the research I've done, it's not discouraged to use the `__str__ ` method to format the error message you'd like to display when the exception is called. When you pass in a message to `super().__init__(message)`, that message is just stored and later outputted by the default `__str__` method.

Reply:
So if we add back the `response`, this no longer is an issue? 

> When you pass in a message to super().__init__(message), that message is just stored and later outputted by the default __str__ method.

This is true now, but may change. E.g. the `PolarisHubError` may at some point add custom logic using the `message` variable. It think using the constructor is the more appropriate way of doing this.

Reply:
This is something we should have by Thursday given the TTL increase, and my time is thin given where this ranks in the priority stack, so let's confirm some things to prevent more confusion:
- We want each error to have a `response` parameter 
- We want to only use the `__init__` method, not the `__str__` method
- We have no use in raising errors with the `from` keyword, and we can raise it like this `raise PolarisHubError("some message")` instead

Is this correct?